nixos/modules/services/web-apps/akkoma.md

```diff
@@ -0,0 +1,136 @@
+referenced file does not exist during startup, unless disabled through
+{option}`services.akkoma.initSecrets`.
+
+The following configuration binds Akkoma to the local IPv6 address `::1` on port 4000, expecting
+to be run behind a HTTP proxy on `fediverse.example.com`.
+
+
+```nix
+services.akkoma.enable = true;
+services.akkoma.config = {
+  ":pleroma" = {
+    ":instance" = {
+      name = "My Akkoma instance";
+      description = "More detailed description";
+      email = "admin@example.com";
+      registration_open = false;
+    };
+
+    "Pleroma.Web.Endpoint" = {
+      url.host = "fediverse.example.com";
+      http = {
+        ip = "::1";
+        port = 4000;
+      };
+    };
+  };
+};
+```
+
+Please refer to the [configuration cheat sheet](https://docs.akkoma.dev/stable/configuration/cheatsheet/)
+for additional configuration options.
+
+The service can be enabled with the usual
+
+```ShellSession
+$ nixos-rebuild switch
+```
+
+## User management {#modules-services-akkoma-user-management}
+
+After the Akkoma service is running, the administration utility can be used to
+[manage users](https://docs.akkoma.dev/stable/administration/CLI_tasks/user/). In particular an
+administrative user can be created with
+
+```ShellSession
+$ sudo -u akkoma RELEASE_COOKIE=/run/akkoma/COOKIE \
+pleroma_ctl user new <nickname> <email> --admin --moderator --password <password>
+```
+
+## Proxy configuration {#modules-services-akkoma-proxy-configuration}
+
+Although it is possible to expose Akkoma directly, it is common practice to operate it behind an
+HTTP reverse proxy such as HAProxy or nginx.
+
+```nix
+security.acme = {
+  email = "admin@example.com";
+  acceptTerms = true;
+};
+
+services.akkoma.nginx = {
+  enableACME = true;
+  forceSSL = true;
+};
+
+services.nginx = {
+  enable = true;
+
+  clientMaxBodySize = "16m";
+  recommendedTlsSettings = true;
+  recommendedOptimisation = true;
+  recommendedGzipSettings = true;
+};
+```
+
+## Frontend management {#modules-services-akkoma-frontend-management}
+
+Akkoma will be deployed with the `pleroma-fe` and `admin-fe` frontends by default. These can be
+overridden by setting the {option}`services.akkoma.frontends` option.
+
+The following example overrides the primary frontend’s default configuration using a custom
+derivation.
+
+```nix
+services.akkoma.frontends.primary.package = let
+  default-config = pkgs.writeText "default-config.json" (builtins.toJSON {
+    expertLevel = 1;
+    collapseMessageWithSubject = false;
+    stopGifs = false;
+    replyVisibility = "following";
+    webPushHideIfCW = true;
+    hideScopeNotice = true;
+    renderMisskeyMarkdown = false;
+    hideSiteFavicon = true;
+    postContentType = "text/markdown";
+    showNavShortcuts = false;
+  });;
+in pkgs.runCommand "pleroma-fe" {
+  nativeBuildInputs = with pkgs; [ jq ];
+} ''
+  cp -R --reflink=auto ${pkgs.akkoma-frontends.pleroma-fe} $out
```

Comment:
Why not use nested symlinks we can modify with lndir?

Reply:
Since the service uses bind mounts to merge the frontends into the state directory, that would work as long as the symlink targets are absolute paths and confinement is disabled.

With confinement enabled, the symlinks would likely point to paths which don’t exist in the mount namespace, unless added explicitly.

Would hard links be a suitable alternative?

```Shell
cp -lR ${pkgs.akkoma-frontends.pleroma-fe} $out
rm $out/static/config.json
jq -s add ${pkgs.akkoma-frontends.pleroma-fe}/static/config.json ${config} >$out/static/config.json
```

Reply:
I don't think that would make much of a difference.